From b5ec929c606ad6cda8a9f92636c8b71fbecf36dc Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Mon, 22 Jun 2020 15:25:27 +0800
Subject: [PATCH] clk: imx: imx8mm: add a rate_count for imx8mm some clocks

add a rate_count for imx8mm some clocks fix an error and keep printing
"clk_pll1416x_set_rate: Invalid rate: 0 for pll clk arm_pll" when
starting

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/clk/imx/clk-imx8mm.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/clk/imx/clk-imx8mm.c b/drivers/clk/imx/clk-imx8mm.c
index 692998a41294..35d4df63d47b 100644
--- a/drivers/clk/imx/clk-imx8mm.c
+++ b/drivers/clk/imx/clk-imx8mm.c
@@ -93,36 +93,43 @@ static const struct imx_pll14xx_rate_table imx8mm_drampll_tbl[] = {
 static struct imx_pll14xx_clk imx8mm_audio_pll __initdata = {
 		.type = PLL_1443X,
 		.rate_table = imx8mm_audiopll_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_audiopll_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_video_pll __initdata = {
 		.type = PLL_1443X,
 		.rate_table = imx8mm_videopll_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_videopll_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_dram_pll __initdata = {
 		.type = PLL_1443X,
 		.rate_table = imx8mm_drampll_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_drampll_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_arm_pll __initdata = {
 		.type = PLL_1416X,
 		.rate_table = imx8mm_pll1416x_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_pll1416x_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_gpu_pll __initdata = {
 		.type = PLL_1416X,
 		.rate_table = imx8mm_pll1416x_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_pll1416x_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_vpu_pll __initdata = {
 		.type = PLL_1416X,
 		.rate_table = imx8mm_pll1416x_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_pll1416x_tbl),
 };
 
 static struct imx_pll14xx_clk imx8mm_sys_pll __initdata = {
 		.type = PLL_1416X,
 		.rate_table = imx8mm_pll1416x_tbl,
+		.rate_count = ARRAY_SIZE(imx8mm_pll1416x_tbl),
 };
 
 static const char *pll_ref_sels[] = { "osc_24m", "dummy", "dummy", "dummy", };
-- 
2.17.1

